File: app/src/main/java/com/castelijns/mmdemo/app/BaseListFragment.java
package com.castelijns.mmdemo.app;

import android.view.View;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;

import com.castelijns.mmdemo.R;

import butterknife.BindView;

public abstract class BaseListFragment extends BaseFragment implements BaseListView {

    @BindView(R.id.tv_header)
    TextView tvHeader;

    @BindView(R.id.pb)
    ProgressBar pb;

    protected void setHeaderText(String text) {
        tvHeader.setText(text);
    }

    @Override
    public void showLoading() {
        pb.setVisibility(View.VISIBLE);
    }

    @Override
    public void hideLoading() {
        pb.setVisibility(View.GONE);
    }

    @Override
    public void showError() {
        Toast.makeText(getContext(), R.string.error_retrieving_data, Toast.LENGTH_SHORT)
                .show();
    }
}

File: app/src/main/kotlin/com/castelijns/mmdemo/app/BaseListFragment.kt
package com.castelijns.mmdemo.app

import android.view.View
import android.widget.Toast
import androidx.fragment.app.Fragment
import com.castelijns.mmdemo.R
import kotlinx.android.synthetic.main.fragment_albums.*

abstract class BaseListFragment : Fragment(), BaseListView {

    protected fun setHeaderText(text: String) {
        tv_header.text = text
    }

    override fun showLoading() {
        pb.visibility = View.VISIBLE
    }

    override fun hideLoading() {
        pb.visibility = View.GONE
    }

    override fun showError() {
        Toast.makeText(context, R.string.error_retrieving_data, Toast.LENGTH_SHORT)
                .show()
    }
}

interface BaseListView {
    fun showLoading()
    fun hideLoading()
    fun showError()
}
